libr/io/p/io_gprobe.c

```diff
@@ -747,13 +752,18 @@ static int gprobe_read (struct gport *port, ut32 addr, ut8 *buf, ut32 count) {
 }
 
 static int gprobe_write (struct gport *port, ut32 addr, const ut8 *buf, ut32 count) {
-	RBuffer *request = r_buf_new ();
-	RBuffer *reply = r_buf_new ();
 	const ut8 cmd = GPROBE_RAM_WRITE_2;
 	ut8 addr_be[4];
 	ut8 count_be[4];
 
-	if (!request || !reply) {
+	RBuffer *request = r_buf_new ();
```

Comment:
you can keep the same code and free both and quit in a single conditional block if you want. will also be correct